test/suites/container_devices_nic_macvlan.sh

```diff
@@ -9,10 +9,14 @@ test_container_devices_nic_macvlan() {
   ip link add "${ct_name}" type dummy
   ip link set "${ct_name}" up
 
+  # Record how many nics we started with.
+  startNicCount=$(find /sys/class/net | wc -l)
+
   # Test pre-launch profile config is applied at launch.
   lxc profile copy default "${ct_name}"
-  lxc profile device set "${ct_name}" eth0 parent "${ct_name}"
-  lxc profile device set "${ct_name}" eth0 nictype "macvlan"
+
+  # Modifiy profile nictype and parent in atomic operation to ensure validation passes.
+  lxc profile show "${ct_name}" | sed  "s/nictype: p2p/nictype: macvlan\n    parent: ${ct_name}/" | lxc profile edit "${ct_name}"
```

Comment:
@stgraber I wanted to draw your attention to this. As part of the validation improvements in this PR, it is no longer supported to be able to change a profile nictype to one that needs a parent without also specifying a parent at the same time. Because the lxc device set tool doesn't currently accept the same param types as lxc device add, i.e multiple key=values separated by a space. In order to get the tests working I've resorted to using the lxc profile edit command and sed to alter both the nictype and the parent in one atomic operation.

We might want to notify users that this is changing so avoid surprise if people have automated this.

Reply:
Indeed we'll want to flag that we're now doing stricter device configuration validation which will catch invalid key combinations but that means that you must now update multiple keys in some situations.

We should also attempt to make the various `set` commands allow the key=value syntax by detecting if all keys include an `=` sign, should be safe to do (no false positives).

Reply:
@stgraber that would be great, I've found myself using the key=var syntax on device set commands frequently.

Reply:
Yeah, I'll send something that does that later.